homeassistant/components/huawei_lte/__init__.py

```diff
@@ -409,7 +414,9 @@ def signal_update() -> None:
         device_registry = await dr.async_get_registry(hass)
         device_registry.async_get_or_create(
             config_entry_id=entry.entry_id,
-            connections=router.device_connections,
+            # We used to have "connections" here before 2021.7, but as of that version
```

Comment:
If we have access to the MAC address, shouldn't we keep it here? It's helpful to be able to link up entities across different integrations.

Reply:
When we get here, we may be in authenticated or unauthenticated mode, depending on configuration, and mac addresses are reliably available only in authenticated mode.

Same applies for the serial number, but it's not a problem for that because we're now guaranteed to have access to it from the config entry, no matter the mode here. But it doesn't seem that we have a place to store the macs in the config entry.

We could grab the macs if in authenticated mode here and pass them along, but I don't know if it would be a problem if we later on another init after a config change end up here in unauthenticated mode. In other words, we'd sometimes call `async_get_or_create` with macs in connections, sometimes without them.

Would that be a problem? If not, I could implement it, but then I'd also need to look into whether having several macs in connections for the same device is ok, as depending on the device, there might be both wifi and ethernet macs. (I guess it wouldn't, but I'd need to make look it up.)

Reply:
You can grab the mac and put it in the device registry right away, or you can store it in the config entry data in case you want to have access to it later.

Reply:
Ok, implemented now. Resulted in a bunch of new code, but hopefully not too much to handle in this PR.